stateupgrade/interfaces.go

```diff
@@ -0,0 +1,35 @@
+// (c) 2023 Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package stateupgrade
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// StateDB is the interface for accessing EVM state in state upgrades
+type StateDB interface {
```

Comment:
we could also move stateupgrade under precompile or reuse interfaces there but I think it is slightly better to keep them separate